src/xmagics/executable.cpp

```diff
@@ -249,19 +250,19 @@ namespace xcpp
 
     void executable::operator()(const std::string& line, const std::string& cell)
     {
-        auto options = get_options();
-        auto parsed = options.parse(line);
-
-        std::string ExeFile = parsed["filename"].as<std::string>();
-        if (ExeFile.empty())
-        {
-            std::cerr << "UsageError: "
-                      << "the following arguments are required: filename"
-                      << std::endl;
-            return;
-        }
+        auto argpars = get_options();
+        argpars.parse(line);
+
+        std::string ExeFile = argpars.get<std::string>("filename");
+//         if (ExeFile.empty())
+//         {
+//             std::cerr << "UsageError: "
+//                       << "the following arguments are required: filename"
+//                       << std::endl;
+//             return;
+//         }
```

Comment:
I guess argparse handles the case where a mandatory argument is missing? If so, this commented code should be removed.

Reply:
Yep, forgot to remove it while cleaning.